File: 2022-9-12/1-two-sum/1-two-sum.js
var twoSum = function(nums, target) {
    var numsObject = {};
    var length=nums.length;
    var i;
    for (i = 0; i < length; i++) {
        if(numsObject[target-nums[i]]!==undefined){
            return [numsObject[target-nums[i]],i];
        }
        numsObject[nums[i]]=i;
    }
}

File: 2022-9-22/1-two-sum/1-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var map = HashMap<Int, Int>()
        for (i in nums.indices) {
            if (map.containsKey(target - nums[i])) {
                return intArrayOf(i, map.getValue(target - nums[i]))
            }
            map[nums[i]] = i
        }
        return IntArray(2)
    }
}